s3/src/test/scala/akka/stream/alpakka/s3/impl/HttpRequestsSpec.scala

```diff
@@ -397,28 +398,29 @@ class HttpRequestsSpec extends AnyFlatSpec with Matchers with ScalaFutures {
     import system.dispatcher
     implicit val materializer: ActorMaterializer = ActorMaterializer()
 
-    val probe = TestProbe()
-    val address = SocketUtil.temporaryServerAddress()
-
-    import akka.http.scaladsl.server.Directives._
+    try {
+      val probe = TestProbe()
+      val address = SocketUtil.temporaryServerAddress()
 
-    Http().bindAndHandle(extractRequestContext { ctx =>
-      probe.ref ! ctx.request
-      complete("MOCK")
-    }, address.getHostName, address.getPort)
+      import akka.http.scaladsl.server.Directives._
 
-    implicit val setting: S3Settings =
-      getSettings().withEndpointUrl(s"http://${address.getHostName}:${address.getPort}/")
+      Http().bindAndHandle(extractRequestContext { ctx =>
+        probe.ref ! ctx.request
+        complete("MOCK")
+      }, address.getHostName, address.getPort)
 
-    val req =
-      HttpRequests.listBucket(location.bucket, Some("random/prefix"), Some("randomToken"))
+      implicit val setting: S3Settings =
+        getSettings().withEndpointUrl(s"http://${address.getHostName}:${address.getPort}/")
 
-    Http().singleRequest(req).futureValue
+      val req =
+        HttpRequests.listBucket(location.bucket, Some("random/prefix"), Some("randomToken"))
 
-    probe.expectMsgType[HttpRequest]
+      Http().singleRequest(req).futureValue shouldBe a[HttpResponse]
 
-    materializer.shutdown()
-    TestKit.shutdownActorSystem(system)
+      probe.expectMsgType[HttpRequest]
+    } finally {
+      TestKit.shutdownActorSystem(system)
+    }
```

Comment:
This fails sometimes when finding a port https://github.com/akka/alpakka/issues/1802